File: Data Structures & Algorithms/three-integer-sum/submission-5.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        nums.sort()
        res = []

        for idx in range(n-2):
            l, r = idx+1, n-1
            if idx > 0 and nums[idx] == nums[idx-1]:
                continue
            while(l < r):
                threesum = nums[idx]+nums[l]+nums[r]
                if threesum == 0:
                    res.append([nums[idx],nums[l],nums[r]])
                    l += 1
                    r -= 1
                    while(l<r and nums[l]==nums[l-1]):
                        l += 1
                elif threesum > 0:
                    r -= 1
                else:
                    l += 1
        return res

            
            

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> um;
        
        for(auto num:nums){
            um[num]++;
        }
        vector<int> init{};
        vector<vector<int>> buckets(nums.size()+1,init);

        for(auto itr=um.begin();itr!=um.end();itr++){
            buckets[itr->second].push_back(itr->first);
        }

        vector<int> result{};
        for(int idx=buckets.size()-1;idx>0;idx--){
            vector<int> bucket=buckets[idx];
            if(bucket.size() > 0 && k > 0){
                for(auto val: bucket){
                    result.push_back(val);
                    k--;
                }
            }
            }
        return result;

    }
};
